code/datums/components/anti_magic.dm

```diff
@@ -17,28 +19,29 @@
 
 	magic = _magic
 	holy = _holy
+	tinfoil = _tinfoil
+	checkslot = _checkslot
 	if(!isnull(_charges))
 		charges = _charges
 	blocks_self = _blocks_self
 	reaction = _reaction
 	expire = _expire
 
 /datum/component/anti_magic/proc/on_equip(datum/source, mob/equipper, slot)
-	if(slot == SLOT_IN_BACKPACK)
+	if((checkslot && slot != checkslot) || slot == SLOT_IN_BACKPACK)
```

Comment:
If you're adding in a custom slot check remove the backpack one. You'll probably need to change some other stuff to make the ones that allow anything but backpack slot work